controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -785,7 +785,7 @@ func (r *BareMetalHostReconciler) manageHostPower(prov provisioner.Provisioner,
 	if desiredPowerOnState {
 		provResult, err = prov.PowerOn()
 	} else {
-		provResult, err = prov.PowerOff()
+		provResult, err = prov.PowerOff(desiredHardReboot)
```

Comment:
This change should be in the second commit - I'd suggest a fixup commit to remove it from this one, then another to add it to the other baremetalhost_controller.go changes (and remove the FIXME comment)

Reply:
OK I think I've done it :-)